core/validatorapi/router.go

```diff
@@ -366,28 +366,28 @@ func proposeBlock(p eth2client.BeaconBlockProposalProvider) handlerFunc {
 				return 0, errors.New("no phase0 block")
 			}
 
-			return proposeBlockResponsePhase0{
-				Version: block.Version,
-				Data:    block.Phase0,
-			}, nil
+			res := new(proposeBlockResponsePhase0)
+			res.Init(block.Phase0)
+
+			return *res, nil
 		case spec.DataVersionAltair:
 			if block.Altair == nil {
 				return 0, errors.New("no altair block")
 			}
 
-			return proposeBlockResponseAltair{
-				Version: block.Version,
-				Data:    block.Altair,
-			}, nil
+			res := new(proposeBlockResponseAltair)
+			res.Init(block.Altair)
```

Comment:
rather write a single function `func marshalBlock(block) []byte, error` that does all this stuff inline. No need for constructors, since we only ever use this one function.